### PR TITLE
CYC-140 Add tests for Show and Index function for SaleController

### LINKED ISSUE #157: CYC-140 Allow Index on sale to be filtered by status
See [CYC-140](https://soen390team13.atlassian.net/browse/CYC-140)
Extend the existing index function to allow filtering by a specific status

### PR DESCRIPTION
**Pull request by RobertoNittolo**
## Description
[CYC-140](https://soen390team13.atlassian.net/browse/CYC-140)
Closes #157 
## Changes
- Add tests for Show and Index function for SaleController

### CODE REVIEW

File: tests/Feature/SaleTest.php [hunk 1]
```diff
@@ -213,4 +213,63 @@ public function test_sales_can_be_updated()
...
+    public function test_sales_can_be_shown()
+    {
+
+        // Create a sale
+        $sale = Sale::create([
+            'description' => "This items will be a test sale",
+            'client_name' => "test name",
+            'status' => "test status",
+            'payment_type' => "test payment",
+            'card_number' => "12341234123412349876987698769876",
```
Comment (luigibesani) [inline tests/Feature/SaleTest.php]: Good work though some of the values you set for the Sale fields are invalid.
- `status` field can only contain : pending, shipped ,paid, or cancelled (Case-sensitive)
- `payment_type` can only contain : Visa or Master Card (Case-sensitive)
- `card_number` can only contain : 16 numeric digits

Comment (luigibesani) [reply tests/Feature/SaleTest.php]: Example :
'status' => "pending",
'payment_type' => "Visa",
'card_number' => "1234123412341234",

Comment (RobertoNittolo) [reply tests/Feature/SaleTest.php]: Nice catch! Changes were made!

**Review by m-triassi — APPROVED** [review]
Perfect!! Good work! and thank you so much @luigibesani for reviewing as well!! ⭐

**Pull request merged**